code/run_capsule.py

```diff
@@ -874,6 +941,38 @@
     with (results_folder / "processing.json").open("w") as f:
         f.write(processing.json(indent=3))
 
+
+    now = datetime.now()
+    # make from scratch:
+    data_description_dict = {}
+    data_description_dict["creation_time"] = now.time()
+    data_description_dict["creation_date"] = now.date()
+    data_description_dict["input_data_name"] = session_name
+    data_description_dict["institution"] = Institution.AIND
+    data_description_dict["investigators"] = []
+    data_description_dict["funding_source"] = [Funding(funder="AIND")]
+    data_description_dict["modality"] = [Modality.ECEPHYS]
+    data_description_dict["experiment_type"] = ExperimentType.ECEPHYS
+    data_description_dict["subject_id"] = session_name.split("_")[1]
+
+    # construct data_description.json
+    if data_description is not None:
+        print("Constructing derived data description from existing data description")
+        existing_data_description_dict = data_description.dict()
+        skip_keys = ["schema_version", "version", "data_level", "described_by", "ror_id",
+                     "creation_time", "creation_date", "institution"]
```

Comment:
@dyf @saskiad can you take a look at this? We have to make sure it gets updated correctly with new `aind-data-schema` version

Reply:
@dyf could you point me to the code to deal with the different `aind-data-schema` versions?

Reply:
You should pull `institution`, `funding_source`, and `modality` from the input. 

You will need to check the input `DataDescription.schema_version` to make sure it works with your version of the schema. The most disruptive change happened in `0.4.0`, when `Institution` become an object (formerly a plain string), modality became a list of objects (formerly a string), and we introduced `experiment_type`.  I think you should be fine if you special case for versions < `0.4.0`. Here is an example:

https://github.com/AllenNeuralDynamics/aind-data-migration-scripts/blob/23cda29eadeced047383053683b5e39e50c5ffd3/scripts/2023-06-08/fix_u19_smartspim.py#L88